keystore/src/mls.rs

```diff
@@ -192,43 +193,17 @@ impl CryptoKeystoreMls for crate::connection::Connection {
 
     #[cfg(not(target_family = "wasm"))]
     async fn mls_fetch_keypackages<V: MlsEntity>(&self, count: u32) -> CryptoKeystoreResult<Vec<V>> {
-        let mut db = self.conn.lock().await;
-
-        let transaction = db.transaction()?;
-
-        let mut stmt = transaction.prepare_cached("SELECT rowid FROM mls_keypackages ORDER BY rowid DESC LIMIT ?")?;
-
-        let mut keypackages: Vec<MlsKeyPackage> = vec![];
-        for kpb_rowid in stmt.query_map([count], |r| r.get(0))? {
-            use std::io::Read as _;
-            let rowid = kpb_rowid?;
-            let mut blob = transaction.blob_open(
-                rusqlite::DatabaseName::Main,
-                "mls_keypackages",
-                "keypackage_ref",
-                rowid,
-                true,
-            )?;
-            let mut keypackage_ref = vec![];
-            blob.read_to_end(&mut keypackage_ref)?;
-            blob.close()?;
-
-            let mut blob = transaction.blob_open(
-                rusqlite::DatabaseName::Main,
-                "mls_keypackages",
-                "keypackage",
-                rowid,
-                true,
-            )?;
-            let mut keypackage = vec![];
-            blob.read_to_end(&mut keypackage)?;
-            blob.close()?;
-
-            keypackages.push(MlsKeyPackage {
-                keypackage_ref,
-                keypackage,
-            });
-        }
+        let mut db_connection = self.conn.lock().await;
+
+        let keypackages = MlsKeyPackage::find_all(
+            &mut db_connection,
+            EntityFindParams {
+                limit: Some(count),
+                offset: None,
+                reverse: true,
+            },
+        )
+        .await?;
```

Comment:
Nice!
BTW, why do we use `reverse: true` in params?

Reply:
To stay consistent with the query we had before. It adds `DESC` after the `ORDER BY` clause.